synthesis/codon/codon.go

```diff
@@ -91,15 +91,39 @@ type Table interface {
 	GetAminoAcids() []AminoAcid
 	GetStartCodons() []string
 	GetStopCodons() []string
+	GetStats() *Stats
 	IsEmpty() bool
 	OptimizeTable(string) Table
 }
 
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
+}
+
+// Add merges the given stats with this instance
+func (s *Stats) Add(toAdd *Stats) {
+	for codon, count := range toAdd.StartCodonCount {
+		s.StartCodonCount[codon] = s.StartCodonCount[codon] + count
+	}
+}
```

Comment:
`Add` should getting added to as we add example protein sequences to the codon table describing the usage of different codons. This would not be a feature of the codon table itself, but something that gets modified over time.